### ISSUE TITLE
Installation problem

### ISSUE THREAD

**Issue by ZarulSaurus** [hallucigenia-sparsa/seqtime#22]
Hello seqtime developers,

There is a problem with installing seqtime. I am using R version 3.5 in RStudio, in Linux:

>\> version
               _                           
platform       x86_64-pc-linux-gnu         
arch           x86_64                      
os             linux-gnu                   
system         x86_64, linux-gnu           
status                                     
major          3                           
minor          5.1                         
year           2018                        
month          07                          
day            02                          
svn rev        74947                       
language       R                           
version.string R version 3.5.1 (2018-07-02)
nickname       Feather Spray

>\>lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty

The error message that I have is this:

>"Installing package into ‘/home/monash/R/x86_64-pc-linux-gnu-library/3.5’
(as ‘lib’ is unspecified)
also installing the dependencies ‘gmp’, ‘partitions’
trying URL 'https://cloud.r-project.org/src/contrib/gmp_0.5-13.2.tar.gz'
Content type 'application/x-gzip' length 131313 bytes (128 KB)
==================================================
downloaded 128 KB
trying URL 'https://cloud.r-project.org/src/contrib/partitions_1.9-19.tar.gz'
Content type 'application/x-gzip' length 374216 bytes (365 KB)
==================================================
downloaded 365 KB
trying URL 'https://cloud.r-project.org/src/contrib/untb_1.7-3.tar.gz'
Content type 'application/x-gzip' length 3702720 bytes (3.5 MB)
==================================================
downloaded 3.5 MB
*installing *source* package ‘gmp’ ...
** package ‘gmp’ successfully unpacked and MD5 sums checked
creating cache ./config.cache
checking for __gmpz_ui_sub in -lgmp... no
configure: error: GNU MP not found, or not 4.1.4 or up, see http://gmplib.org
ERROR: configuration failed for package ‘gmp’
*removing ‘/home/monash/R/x86_64-pc-linux-gnu-library/3.5/gmp’
Warning in install.packages :
  installation of package ‘gmp’ had non-zero exit status
ERROR: dependency ‘gmp’ is not available for package ‘partitions’
*removing ‘/home/monash/R/x86_64-pc-linux-gnu-library/3.5/partitions’
Warning in install.packages :
  installation of package ‘partitions’ had non-zero exit status
ERROR: dependency ‘partitions’ is not available for package ‘untb’
*removing ‘/home/monash/R/x86_64-pc-linux-gnu-library/3.5/untb’
Warning in install.packages :
  installation of package ‘untb’ had non-zero exit status
The downloaded source packages are in
	‘/tmp/RtmphhOU2v/downloaded_packages’"

Can you help me with this? Thank you.

Regards,
Zarul

**Comment by hallucigenia-sparsa** [comment]
The reason for this problem appears to be a failure to install the "gmp" package on ubuntu. Does installing "gmp" separately from its source fix this problem?

**Comment by ZarulSaurus** [comment]
Yes, it was a gmp issue. Turns out I did not have a GMP C library. This solved it:

`sudo apt-get install libgmp-dev`

Thank you very much!

**Comment by Lin936** [comment]
Hello seqtime developers,
I have got a problem with installing seqtime :

Downloading GitHub repo hallucigenia-sparsa/seqtime@HEAD
√  checking for file 'C:\Users\user\AppData\Local\Temp\RtmpmIeDhQ\remotes161a044df3709\hallucigenia-sparsa-seqtime-67a93ed/DESCRIPTION' (383ms)
-  preparing 'seqtime':
√  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts (398ms)
-  checking for empty or unneeded directories
-  building 'seqtime_0.1.1.tar.gz'
   
* installing *source* package 'seqtime' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
錯誤: (從警告轉換成) package 'igraph' was built under R version 4.0.2
停止執行
ERROR: lazy loading failed for package 'seqtime'
* removing 'C:/Program Files/R/R-4.0.1/library/seqtime'
installation of package ��C:/Users/user/AppData/Local/Temp/RtmpmIeDhQ/file161a02359290e/seqtime_0.1.1.tar.gz�� had non-zero exit status

Thanks for your help
Rgards,
Lin

**Comment by antagomir** [comment]
Hi - have you tried devtools::install_github("hallucigenia-sparsa/seqtime")

**Comment by Lin936** [comment]
Hi Prof. Lahti,
I tried just now,
Still saw the same error:

> devtools::install_github("hallucigenia-sparsa/seqtime")Downloading GitHub repo hallucigenia-sparsa/seqtime@HEAD√  checking for file 'C:\Users\user\AppData\Local\Temp\RtmpmIeDhQ\remotes161a05fe6d8a\hallucigenia-sparsa-seqtime-67a93ed/DESCRIPTION' (589ms)-  preparing 'seqtime': (379ms)√  checking DESCRIPTION meta-information ... -  checking for LF line-endings in source and make files and shell scripts (504ms)-  checking for empty or unneeded directories-  building 'seqtime_0.1.1.tar.gz'

* installing *source* package 'seqtime' ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
錯誤: (從警告轉換成) package 'igraph' was built under R version 4.0.2
停止執行
ERROR: lazy loading failed for package 'seqtime'
* removing 'C:/Program Files/R/R-4.0.1/library/seqtime'installation of
package ��C:/Users/user/AppData/Local/Temp/RtmpmIeDhQ/file161a055e53ae/seqtime_0.1.1.tar.gz��
had non-zero exit status

**Comment by antagomir** [comment]
Ok. This seems to work on my Ubuntu desktop. Windows is often more difficult regarding R package operations. You may be lacking admin permissions to install packages. Perhaps @hallucigenia-sparsa could help here.

**Comment by Lin936** [comment]
Thanks Prof. Lahti!
